src/atom_ftyp.cpp

```diff
@@ -53,7 +53,7 @@ void MP4FtypAtom::Generate()
 
 void MP4FtypAtom::Read()
 {
-   if ( m_size == 0ULL )
+   if ( m_size == 0ULL || m_size < 8ULL )
```

Comment:
Nitpick, but wouldn't `if ( m_size < 8ULL )` have sufficed? 🤔

Reply:
Good question.  With optimization disabled it does make two (`2`) condition checks:
![image](https://github.com/user-attachments/assets/15b3045c-be4e-4952-aa16-7242f9552d74)

But with optimizations enabled it is optimized away:
![image](https://github.com/user-attachments/assets/1e142b89-e1ca-49df-a942-2b55439e8166)

(Online link: https://godbolt.org/z/bPjM4fcds )

So I interpreted it as readability reasons instead of succinctness reasons.

Hope that helps! :smiley: